### ISSUE TITLE
RAF polyfill interferes with Angular Zone

### ISSUE THREAD

**Issue by chenxinyanc** [juggle/resize-observer#80]
**Describe the bug**
I'm using this package as a polyfill for `window.ResizeObserver` when the native one is not available. However, this package will *unconditionally* override the global RAF callback, which may conflict with the RAF override provided in [Angular zone.js](https://github.com/angular/angular/tree/master/packages/zone.js) package, causing the loss of zone information.

The underlying `requestAnimationFrame` is called only for the first RAF call during a frame, and all the queued RAF callbacks are executed in the same "zone" (i.e. the "zone" of first RAF callback.)

https://github.com/juggle/resize-observer/blob/0d8bc59b39cbf46454387b379773052860c0c81e/src/utils/scheduler.ts#L133-L139

I understand working with multiple polyfills can be tricky. However, I think it would be better if you do not apply the RAF polyfill when importing the module, but when the first `ResizeObserver` instance from this library is created. In this case, at least I can ensure RAF is not replaced when I'm using Browser's native `ResizeObserver` (e.g. on Chrome).

**To Reproduce**
See https://angular-cqmqgl.stackblitz.io/ . Open the console to see the error. See the **Expected behavior** part for full project.

![image](https://user-images.githubusercontent.com/50703224/66369054-57ac7c00-e9cd-11e9-8f52-f679474efd07.png)

Basically, what I have done is:
* Start an Angular application.
* Import the `@juggle/resize-observer` module.
* Call `requestAnimationFrame` both inside and outside NgZone, in a way that the RAF callbacks are executed in the same frame. Check if we are inside Angular zone in the RAF callbacks with `NgZone.assertInAngularZone` and `NgZone.assertNotInAngularZone()`.
* Sample code to reproduce the bug is like this
```ts
import { Component, NgZone } from '@angular/core';
import { ResizeObserver as ResizeObserverPolyfill } from '@juggle/resize-observer';

@Component({
  selector: 'my-app',
  templateUrl: './app.component.html',
  styleUrls: [ './app.component.css' ]
})
export class AppComponent  {
  name = 'Angular';
  myText = String(ResizeObserverPolyfill);
  constructor(private _zone: NgZone) {
  }
  ngOnInit() {
    // RAF callback inside NgZone
    rafCallback(this._zone, true, true);
    // RAF callback outside NgZone
    rafCallback(this._zone, false, true);
  }
}

function rafCallback(zone: NgZone, isInZone: boolean, suppressAssertion?: boolean) {
  if (isInZone) {
    suppressAssertion || NgZone.assertInAngularZone();
    zone.run(() => window.requestAnimationFrame(() => rafCallback(zone, isInZone)));
  } else {
    suppressAssertion || NgZone.assertNotInAngularZone();
    zone.runOutsideAngular(() => window.requestAnimationFrame(() => rafCallback(zone, isInZone)));
  }
}
```

**Expected behavior**
The `assertNotInAngularZone` call should pass.

**Sample Code**
See https://stackblitz.com/edit/angular-cqmqgl .

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser: Chromium Edge
 - Version: 79.0.287.3 

**Smartphone (please complete the following information):**
 - Device: 
 - OS: 
 - Browser: 
 - Version: 

**Additional context**

**Comment by TremayneChrist** [comment]
Hi @chenxinyanc thanks for your detailed report!
The next version will remove this. You can see the WIP here #77.

I'll try and get this moving again, so that it'll fix this for you :)

**Comment by TremayneChrist** [comment]
Fixed in v2.4.0
